### DISCUSSION TITLE
Lab 4 Part 3 Test 3 - Consider including ShardMaster in active partition

### DISCUSSION THREAD

**Discussion by rtvu** [emichael/dslabs#47] [Ideas]
My implementation for transactions includes communicating with ShardMaster. Due to the partition in this test, I had to special-case single group transactions in order to pass.

I was wondering if this test can be relaxed to include ShardMaster in the active partition.

Clearly, there are trade-offs in including another server in the critical path, however, including ShardMaster also opens up interesting design decisions and potential benefits.

**Comment by emichael** [comment]
It's an interesting idea, but it's not really in the spirit of the assignment. If you put the ShardMaster cluster on the critical path for every transaction, then the system is no longer scalable. Also, I don't really think that communicating with the ShardMaster first buys you much, since an arbitrary amount of time and number of reconfigurations could go by between the ShardMaster's replies and their receipt. Unless the ShardMaster itself is acting as transaction coordinator, which it absolutely should not be doing.

Perhaps this could be better documented in the lab's README, though.
